RT/Renderer/Backend/DX12/src/ImageReadWrite.cpp

```diff
@@ -308,11 +325,33 @@ RT_Image RT_LoadDDSFromDisk(RT_Arena *arena, RT_String path)
 	// Don't want to spam when attempting to load DDS image first
 	// fprintf(stderr, "[RT_LoadDDSFromDisk]: Attempting to load image: '%.*s'\n", RT_ExpandString(path));
 
+	
+	//attempt to load from vaults first here;
+	bool loaded = false;
+
 	RT_String memory;
-	if (RT_ReadEntireFile(arena, path, &memory))
+	memory.bytes = nullptr;
+	memory.count = 0;
+
+	if (RT_GetFileFromVaults(path, memory))
+	{
+
+		loaded = true;
+	}
+
+	
+	else if (RT_ReadEntireFile(arena, path, &memory))
+	{
+		
+		loaded = true;
+	}
+
+	if (loaded)
 	{
 		result = RT_LoadDDSFromMemory(memory);
 	}
 
+	RT_ArenaReset(&g_thread_arena);
```

Comment:
There was a memory leak when loading DDS files (Every time a level loaded it took 1GB+ more memory).  I could not figure out exactly where it was coming from, but adding this reset line took care of the issue.  Not sure if this was proper way to deal with it.   I think I may still have a similar issue with PNG files, but adding a similar reset command at the end of the png load code caused the game to crash.  Once the idea for SGV files is approved I will look deeper into the memory leak.  Unless you have insight into what might be causing it.